lib/Data/Validate/Sanctions/Fetcher.pm

```diff
@@ -126,28 +172,35 @@ sub _process_name_and_dob {
         $dob = $1 if $dob =~ m/^[A-Z][a-z]{2}-(\d{4})$/;
 
         if ($dob =~ m/^\d{4}$/) {
-            push @year_list, $dob;
+            push @dob_year, $dob;
         } elsif ($dob =~ m/(\d{4}).*to.*(\d{4})$/) {
-            push @year_list, ($1 .. $2);
+            push @dob_year, ($1 .. $2);
         } else {
             my $epoch = _date_to_epoch($dob);
-            (defined $epoch) ? push(@epoch_list, $epoch) : push(@other_list, $dob);
+            (defined $epoch) ? push(@dob_epoch, $epoch) : push(@dob_text, $dob);
         }
     }
 
-    for my $name (@$name_list) {
+    delete $data{date_of_birth};
+    $data{dob_epoch} = \@dob_epoch;
+    $data{dob_year}  = \@dob_year;
+    $data{dob_text}  = \@dob_text;
+
+    # convert all country names to iso codes
+    for my $field (qw/place_of_birth residence nationality citizen/) {
+        $data{$field} = [map { my $val = trim $_; lc(code2country($val) ? $_ : country2code($val) // '') } $data{$field}->@*];
```

Comment:
If `code2country($val)` evaluates to false (`undef` I guess), you don't to call it again in the `else` of the ternary operator?

Reply:
It's not called twice. The second one is just the inverse: `country2code`. Country can be a code or name, depending on the data source. I'm converting all of them to codes here.

Reply:
Such a shame, my eyes deceived me. Sorry about that!